### PR TITLE
feat: Rename clientSecret to clientKey

### PR DESCRIPTION
**Pull request by chriswk**

### CODE REVIEW

File: src/test/kotlin/io/getunleash/UnleashConfigTest.kt [hunk 1]
```diff
@@ -54,14 +54,14 @@ class UnleashConfigTest {
...
-            .hasMessage("You have to set client secret in your UnleashConfig")
+            .hasMessage("You have to set client key in your UnleashConfig")
     }
 
     @Test
     fun `Can set client timeouts in seconds with builder methods`() {
         val config = UnleashConfig(
             proxyUrl = "https://localhost:4242/proxy",
-            clientSecret = "some-secret",
+            clientKey = "some-secret",
```
Comment (thomasheartman) [inline src/test/kotlin/io/getunleash/UnleashConfigTest.kt:64]: Very minor, but we might want to update the value here (and in the other tests) to indicate that it's not a secret anymore.

```suggestion
            clientKey = "some-key",
```

File: src/test/kotlin/io/getunleash/polling/AutoPollingPolicyTest.kt [hunk 2]
```diff
@@ -89,7 +89,7 @@ class AutoPollingPolicyTest {
         val server = MockWebServer()
         val isCalled = CompletableFuture<Unit>()
         val pollingMode = PollingModes.autoPoll(20) { isCalled.complete(null) }
-        val config = UnleashConfig.newBuilder().proxyUrl(server.url("/").toString()).clientSecret("my-secret").build()
+        val config = UnleashConfig.newBuilder().proxyUrl(server.url("/").toString()).clientKey("my-secret").build()
```
Comment (thomasheartman) [inline src/test/kotlin/io/getunleash/polling/AutoPollingPolicyTest.kt:92]: The above `key` vs `secret` in variable values applies here too.

**Review by thomasheartman — APPROVED** [review]
Nice! Looks great 😄  Just thinking we should update the value we use in some of the tests. It's definitely minor, but better to just be thorough, I reckon 💁🏼 But I'll leave it up to you in the end. Thanks! 🙏🏼

File: src/test/kotlin/io/getunleash/polling/AutoPollingPolicyTest.kt (hunk 2 above)
Comment (chriswk) [reply src/test/kotlin/io/getunleash/polling/AutoPollingPolicyTest.kt:92]: Fixed in 5f9462e

**Pull request merged**